compiler/noirc_evaluator/src/ssa/ir/function.rs

```diff
@@ -211,6 +211,16 @@ impl Function {
 
         unreachable!("SSA Function {} has no reachable return instruction!", self.id())
     }
+
+    pub(crate) fn num_instructions(&self) -> usize {
+        self.reachable_blocks()
+            .iter()
+            .map(|block| {
+                let block = &self.dfg[*block];
+                block.instructions().len() + block.terminator().is_some() as usize
```

Comment:
Do we need to count the terminator here? It should always be Some after the Ssa is initially constructed so even if we do want to count it, maybe we should just add a constant 1?

Reply:
I do prefer this to a magic number `1`